code/ch02/default_constructor_initializer_list_order.dart

```diff
@@ -0,0 +1,31 @@
+// This code shows that when using the default (no-arg) constructor of
```

Comment:
I wouldn't put it this way, since I think of the default constructor being the one that's created if you don't explicitly create one. Also, it doesn't matter whether the subclass has args or not, and I wouldn't call it the "main" class. Maybe:

// This code shows that when a constructor doesn't explicitly call its
// superclass's constructor, the superclass's no-argument constructor
// is called. Here's the order:
//
// 1. Class's initializer list executes.
// 2. Superclass's no-argument constructor executes.
// 3. Class's constructor executes.